app/audit/models/OptOutAuditModel.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package audit.models
+
+import auth.MtdItUser
+import connectors.itsastatus.ITSAStatusUpdateConnectorModel.{ITSAStatusUpdateResponse, ITSAStatusUpdateResponseFailure, ITSAStatusUpdateResponseSuccess}
+import models.incomeSourceDetails.TaxYear
+import models.itsaStatus.ITSAStatus.ITSAStatus
+import play.api.libs.json.{JsObject, JsValue, Json, OFormat}
+import services.optout.OptOutProposition
+
+case class Outcome(
+                    isSuccessful: Boolean,
+                    failureCategory: Option[String],
+                    failureReason: Option[String]
+                  )
+
+
+object Outcome {
+  implicit val format: OFormat[Outcome] = Json.format[Outcome]
+}
+
+case class OptOutAuditModel(
+                             //                             mtdItUser: MtdItUser[_],
+                             nino: String,
+                             outcome: Outcome,
+                             optOutRequestedFromTaxYear: String,
+                             currentYear: String,
+                             beforeITSAStatusCurrentYearMinusOne: ITSAStatus,
+                             beforeITSAStatusCurrentYear: ITSAStatus,
+                             beforeITSAStatusCurrentYearPlusOne: ITSAStatus,
+                             afterAssumedITSAStatusCurrentYearMinusOne: ITSAStatus,
+                             afterAssumedITSAStatusCurrentYear: ITSAStatus,
+                             afterAssumedITSAStatusCurrentYearPlusOne: ITSAStatus,
+                             currentYearMinusOneCrystallised: Boolean
+                           ) extends ExtendedAuditModel {
+
+
+  override val transactionName: String = enums.TransactionName.OptOutQuarterlyReportingRequest
+
+  override val auditType: String = enums.AuditType.OptOutQuarterlyReportingRequest
+
+  private val optOutDetailsJson: JsObject =
+    Json.obj(
+      "nino" -> nino,
```

Comment:
The TXM Doc also shows `saUtr`, `credId`, `agentReferenceNumber`, `userType` and `mtditid` I think those need to be included in the model as well, do correct me if I'm wrong though

I know some of the other audit models use `userAuditDetails(mtdItUser)` for gathering some of those details which is part of the Utilities.scala file

Reply:
for some reason I had added these then removed them. Think when I was trying to get the Ispecs to work somehow confused myself and talked myself out and ended up removing them. 

Anyways they should be there and will add them back in

Reply:
decided to just added them as fields not the mtduser then we can just from the request/implicit mtduser pull out the data and populate the fields, makes formatting to json easier if it's ever needed.